puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/packaging.groovy

```diff
@@ -0,0 +1,61 @@
+def packages_to_build
+
+pipeline {
+    agent { label 'rpmbuild' }
+
+    options {
+        timestamps()
+        timeout(time: 2, unit: 'HOURS')
+        ansiColor('xterm')
+        buildDiscarder(logRotator(numToKeepStr: '15'))
+    }
+
+    stages {
+        stage('Clone Packaging') {
+            steps {
+
+                deleteDir()
+                checkout changelog: true, poll: false, scm: [
+                    $class: 'GitSCM',
+                    branches: [[name: '${ghprbActualCommit}']],
+                    doGenerateSubmoduleConfigurations: false,
+                    extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                    submoduleCfg: [],
+                    userRemoteConfigs: [
+                        [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/${ghprbGhRepository}']
+                    ]
+                ]
+
+            }
+        }
+
+        stage('Find Packages to Build') {
+            steps {
+
+                script {
+                    changed_packages = sh(returnStdout: true, script: "git diff origin/${ghprbTargetBranch} --name-only -- 'packages/**.spec' | cut -d'/' -f2 |sort -u").trim()
```

Comment:
Rather than cut, would dirname + basename be possible? That wouldn't assume a directory structure making it work for older branches too.